optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/persistence/ConferenceSchedulingImporter.java

```diff
@@ -0,0 +1,511 @@
+        if (Arrays.asList(smallRoomsTypeNames).contains(talkTypeName)) {
+            preferredRoomTagSet.add(SMALL_ROOM_TAG);
+        } else if (Arrays.asList(mediumRoomsTypeNames).contains(talkTypeName)) {
+            preferredRoomTagSet.add(MEDIUM_ROOM_TAG);
+        } else {
+            preferredRoomTagSet.add(LARGE_ROOM_TAG);
+        }
+        return preferredRoomTagSet;
+    }
+
+    private Set<String> getProhibitedRoomTagSetForTalkOfType(String talkTypeName) {
+        Set<String> prohibitedRoomTagSet = new HashSet<>();
+        if (Arrays.asList(largeRoomsTypeNames).contains(talkTypeName)) {
+            prohibitedRoomTagSet.add(SMALL_ROOM_TAG);
+            prohibitedRoomTagSet.add(MEDIUM_ROOM_TAG);
+        } else if (Arrays.asList(mediumRoomsTypeNames).contains(talkTypeName)) {
+            prohibitedRoomTagSet.add(SMALL_ROOM_TAG);
+        }
+
+        return prohibitedRoomTagSet;
+    }
+
+    private Set<TalkType> getTalkTypeSetForCapacity(int capacity) {
+        Set<TalkType> talkTypeSet = new HashSet<>();
+        List<String> typeNames = new ArrayList<>();
+        if (capacity < 100) {
+            typeNames.addAll(Arrays.asList(smallRoomsTypeNames));
+        } else if (capacity < 1000) {
+            typeNames.addAll(Arrays.asList(mediumRoomsTypeNames));
+            typeNames.addAll(Arrays.asList(smallRoomsTypeNames));
+        } else {
+            typeNames.addAll(Arrays.asList(largeRoomsTypeNames));
+            typeNames.addAll(Arrays.asList(mediumRoomsTypeNames));
+            typeNames.addAll(Arrays.asList(smallRoomsTypeNames));
+        }
+
+        for (String talkTypeName : typeNames) {
+            TalkType talkType = talkTypeNameToTalkTypeMap.get(talkTypeName);
+            if (talkType != null) {
+                talkTypeSet.add(talkType);
+            }
+        }
+
+        return talkTypeSet;
+    }
+
+    private JsonObject readJsonObject(String url) {
+        try (InputStream inputStream = openConnectionCheckRedirects(new URL(url).openConnection())) {
+            JsonReader jsonReader = Json.createReader(inputStream);
+            return jsonReader.readObject();
+        } catch (IOException e) {
+            throw new IllegalStateException(
+                    "Import failed on URL (" + url + ").", e);
+        }
+    }
+
+    private JsonArray readJsonArray(String url) {
+        try (InputStream inputStream = openConnectionCheckRedirects(new URL(url).openConnection())) {
+            JsonReader jsonReader = Json.createReader(inputStream);
+            return jsonReader.readArray();
+        } catch (IOException e) {
+            throw new IllegalStateException(
+                    "Import failed on URL (" + url + ").", e);
+        }
+    }
+
+    private InputStream openConnectionCheckRedirects(URLConnection connection) throws IOException { // credits for https://www.cs.mun.ca/java-api-1.5/guide/deployment/deployment-guide/upgrade-guide/article-17.html
```

Comment:
This is a very general method, suitable for some utility class. There is no reason `ConferenceSchedulingImporter` should care about details how to follow http(s) redirects.

Reply:
It would also be nice to have a unit test for that.  E.g. http://www.mock-server.com/ might help with that - you can easily mock expected response from a server.